lib/xapi-stdext-pervasives/pervasiveext.ml

```diff
@@ -15,14 +15,23 @@
  * Even if fct raises an exception, clean_f is applied
 *)
 
+let src = Logs.Src.create "pervasiveext" ~doc:"logs from Xapi_stdext_pervasives.Pervasiveext"
+
 
 let finally fct clean_f =
   let result =
     try
       fct ();
     with exn ->
       Backtrace.is_important exn;
-      clean_f ();
+      begin
+        (* We catch and log exceptions raised by clean_f to avoid shadowing
+           the original exception raised by fct *)
+        try
+          clean_f ();
+        with cleanup_exn ->
+          Logs.err ~src (fun m -> m "finally: Error while running cleanup after failure of main function: %s" (Printexc.to_string cleanup_exn));
```

Comment:
According to the level usage conventions error is too high here, we do not know that this is causing errors in the program (the caller might be ignoring all errors): http://erratique.ch/software/logs/doc/Logs.html#usage
Perhaps  warning would be more suitable, we do not expect the cleanup function to fail.

Reply:
You could also use `%a` and `Fmt.exn` instead of Printexc.to_string, but its fine either way.

Reply:
Thanks, updated